actioncable/lib/action_cable.rb

```diff
@@ -35,7 +35,8 @@ module ActionCable
       confirmation: 'confirm_subscription'.freeze,
       rejection: 'reject_subscription'.freeze
     },
-    default_mount_path: '/cable'.freeze
+    default_mount_path: '/cable'.freeze,
+    protocols: ["actioncable-v1-json".freeze, "actioncable-unsupported".freeze].freeze
```

Comment:
@javan Yeah the reason for this is that the browser side WebSocket client sticks with the bizarre recommended implementation where the client does not tell us why it disconnected, when it could either be a bad disconnect or a protocol mismatch. So the `actioncable-unspported` protocol is the way to get around this: it will always be the last to be chosen, and if it is, we know for sure there is a mismatch. The alternatives to this are really hacky.

@maclover7 I initially asked myself the same thing. However, the protocol version is distinct from the Rails version, and ideally it would be changed quite a bit less often than the Rails version. Also since these protocol changes are almost always breaking changes, semantic versioning is not quite appropriate here.